### ISSUE TITLE
Code conflicts when gophercloud is updated to v0.24.0

### ISSUE THREAD

**Issue by hs0210** [openshift/installer#5653]
# Platform:

IPI

# What happened?

When I updated https://github.com/openshift/installer/pull/5196 and https://github.com/openshift-metal3/terraform-provider-ironic/pull/53 based on https://github.com/metal3-io/baremetal-operator/pull/1063 (for feature [Support RAID and BIOS configuration for baremetal IPI deployments](https://github.com/openshift/enhancements/pull/791)), some dependencies of installer need to be updated to a higher version. The main change is `gophercloud v0.17.0 => v0.22.0`,  which causes serious dependency conflicts, thus a lot of dependencies have to be upgraded in order to compile successfully. But the compiled binary encountered the following error when executing:
```
panic: proto: file "tfplugin5.proto" is already registered
See https://developers.google.com/protocol-buffers/docs/reference/go/faq#namespace-conflict


goroutine 1 [running]:
google.golang.org/protobuf/reflect/protoregistry.glob..func1({0x16491b20, 0x16a2d328}, {0x16491b20, 0xc0016658f0})
        /home/kni/hs/installer/vendor/google.golang.org/protobuf/reflect/protoregistry/registry.go:54 +0x1f4
google.golang.org/protobuf/reflect/protoregistry.(*Files).RegisterFile(0xc0004fd038, {0x16a2d328, 0xc000388e00})
        /home/kni/hs/installer/vendor/google.golang.org/protobuf/reflect/protoregistry/registry.go:128 +0x399
google.golang.org/protobuf/internal/filedesc.Builder.Build({{0x11251cc2, 0x45}, {0x2584f820, 0x1e73, 0x1e73}, 0x3, 0x3a, 0x0, 0x2, {0x164f1ee0, ...}, ...})
        /home/kni/hs/installer/vendor/google.golang.org/protobuf/internal/filedesc/build.go:113 +0x1d6
google.golang.org/protobuf/internal/filetype.Builder.Build({{{0x11251cc2, 0x45}, {0x2584f820, 0x1e73, 0x1e73}, 0x3, 0x3a, 0x0, 0x2, {0x0, ...}, ...}, ...})
        /home/kni/hs/installer/vendor/google.golang.org/protobuf/internal/filetype/build.go:139 +0x198
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5.file_tfplugin5_proto_init()
        /home/kni/hs/installer/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5.pb.go:4603 +0x258
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5.init.0()
        /home/kni/hs/installer/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5.pb.go:3917 +0x17
```

The versions of the following modules have changed from before:
1. github.com/gophercloud/gophercloud v0.17.0 => v0.24.0
2. github.com/gophercloud/utils v0.0.0-20210720165645-8a3ad2ad9e70 => v0.0.0-20220209210848-d0ab9f2a8909
3. github.com/hashicorp/go-plugin v1.3.0 => v1.4.3
4. github.com/terraform-provider-openstack/terraform-provider-openstack v1.37.0 => v1.47.0
5. github.com/zclconf/go-cty v1.8.1 => v1.9.1
6. golang.org/x/crypto v0.0.0-20210817164053-32db794688a5 => v0.0.0-20211202192323-5770296d904e
7. golang.org/x/net v0.0.0-20210825183410-e898025ed96a => v0.0.0-20211112202133-69e39bad7dc2
8. **github.com/hashicorp/terraform-plugin-sdk/v2 v2.10.0 (newly introduced)**
9. github.com/Microsoft/go-winio v0.4.16 (newly introduced)
10. github.com/hashicorp/go-getter v1.5.2 => v1.5.3
11. github.com/hashicorp/go-hclog v0.14.1 => v0.16.1
12. github.com/hashicorp/go-multierror v1.1.0 => v1.1.1
13. github.com/hashicorp/go-cty v1.4.1-0.20200414143053-d3edf31b6320 (newly introduced)
14. github.com/hashicorp/terraform-exec v0.13.1 => v0.15.0
15. github.com/hashicorp/terraform-json v0.9.0 => v0.13.0
16. **github.com/hashicorp/terraform-plugin-go v0.5.0 (newly introduced)**
17. github.com/hashicorp/terraform-plugin-log v0.2.0 (newly introduced)
18. github.com/hashicorp/terraform-registry-address v0.0.0-20210412075316-9b2996cce896 (newly introduced)
19. github.com/mitchellh/copystructure v1.0.0 => v1.2.0
20. github.com/mitchellh/reflectwalk v1.0.1 => v1.0.2
21. github.com/vmihailenco/msgpack v4.0.4+incompatible (newly introduced)
22. github.com/xanzy/ssh-agent v0.2.1 => v0.3.0

The newly introduced modules in bold above will be working together with the existing `hashicorp-terraform-plugin-sdk` module:
1. github.com/openshift/hashicorp-terraform-plugin-sdk:v1.16.0-openshift (sdk/v1 + `old tfplugin5`)
2. github.com/hashicorp/terraform-plugin-sdk/v2:v2.10.0
3. github.com/hashicorp/terraform-plugin-go:v0.5.0 (`new tfplugin5`)

It seems that the panic is thrown due to the simultaneous use of the `old tfplugin5` and the `new tfplugin5`

# What you expected to happen?

The `openshift-install` works fine.

# How to reproduce it (as minimally and precisely as possible)?

1. Use https://github.com/hs0210/installer/tree/update-gophercloud branch to compile the `openshift-install`
2. Execute `./bin/openshift-install version`

```console
$ TAGS="libvirt baremetal" hack/build.sh
$ ./bin/openshift-install version
```

**Comment by hs0210** [comment]
#### Background Statement:

When I updated https://github.com/openshift/installer/pull/5196 and https://github.com/openshift-metal3/terraform-provider-ironic/pull/53 to the latest state to test features, I ran into a conflict when compiling the installer.

The conflict is caused by too much difference between the gophercloud versions vendored by BMO(v0.22.0) and the installer(v0.17.0). After I updated gophercloud to the latest v0.24.0 (also to resolve some other dependency conflicts) and resolved the dependency conflicts by updating some other vendor packages, the installer can compile successfully, but when executing `openshift-install --dir ~/clusterconfigs create manifests`, the above panic occurred. 

Could someone give me a hint about how to solve this problem, thanks a lot!

**Comment by hs0210** [comment]
#### About what we have tried:

We noticed that `openshift/hashicorp-terraform-plugin-sdk` is forked from `hashicorp/terraform-plugin-sdk` and [makes tfplugin5 packages public](https://github.com/openshift/hashicorp-terraform-plugin-sdk/pull/4/commits/f6eac732761894b3ad98db99f43cef9b0873b1d5), while `hashicorp/terraform-plugin-sdk` itself has [removed tfplugin5 package](https://github.com/hashicorp/terraform-plugin-sdk/commit/5c7be201e9e532dffa976505a7ac4c0dd76e0490) since v2.2.0, and said that *terraform-plugin-go is now the common interface for the Terraform protocol*.

So we tried to remove the tfplugin5 from`openshift/hashicorp-terraform-plugin-sdk` and updated whole project to only use the tfplugin5 provided by `terraform-plugin-go`. But the tfplugin5 has been changed a lot from the old one, so we encountered conflicts again while compiling...

Anyway, above is just for your information, I am not sure if this is the right direction.

**Comment by hs0210** [comment]
Fixed by #5507
